### PR TITLE
Add card parameter source

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/26898
Demo https://www.loom.com/share/b7de0812f27648dc9326a5ce21e95602
Depends on https://github.com/metabase/metabase/pull/27343

How to verify:
- Add native query `select distinct CATEGORY from PRODUCTS order by CATEGORY`
- Add a dashboard and a question based on products
- Edit dashboard -> Filter -> Text or Category -> Dropdown
- Map this parameter to the question on the dashboard
- Select parameter -> Options to pick from -> From question
- Select the native question and its category column and save the dashboard
- Make sure you can filter the dashboard with dropdown values coming from the question

Notes:
- Only string columns are supported

Not yet implemented:
- Search for questions on the first step
- Empty state for the second step
- Values preview
- Filtering on data types, e.g. for not allowing to select tables. Requires `DataPicker` changes.

Screenshot 2022-12-21 at 22 58 31
Screenshot 2022-12-21 at 22 58 46
Screenshot 2022-12-21 at 22 58 53

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/27354)

### CODE REVIEW

File: frontend/src/metabase/components/ListField/ListField.tsx [hunk 1]
```diff
@@ -129,8 +129,8 @@ const ListField = ({
       )}
 
       <OptionsList isDashboardFilter={isDashboardFilter}>
-        {filteredOptions.map(option => (
-          <OptionContainer key={option[0]}>
+        {filteredOptions.map((option, index) => (
+          <OptionContainer key={index}>
```
Comment (ranquild) [inline frontend/src/metabase/components/ListField/ListField.tsx:133]: Required to work with duplicate values

File: frontend/src/metabase/entities/containers/EntityObjectLoader.jsx [hunk 2]
```diff
@@ -31,6 +32,7 @@ const getMemoizedEntityQuery = createMemoizedSelector(
 
 class EntityObjectLoaderInner extends React.Component {
   static defaultProps = {
+    requestType: "fetch",
```
Comment (ranquild) [inline frontend/src/metabase/entities/containers/EntityObjectLoader.jsx:35]: This change is needed to use `fetchMetadata` via entity loaders

**Comment by WiNloSt** [comment]
These are some of the behaviors I found when testing that might need to be addressed.

1. Moving a source question to a different collection sometimes would crash the parameter source picker.
    1. Have the source question on a different collection
    2. Edit the dashboard, click **Values from a model or question**, and click the collection that you want to move the source question into (this is crucial).
    3. Close the modal.
    4. Move the question to the destined collection (the one that you clicked in step 2)
    5. Repeat step 2 to see the error.
    https://www.loom.com/share/e663c2a66aac449e9a2b951be2bcc87c
2. Moving a source question to a different collection sometimes would create duplicate entries in the parameter source picker.
    1. Have the source question somewhere.
    2. Edit the dashboard, click **Values from a model or question**, and click the collection that contains that source question (this is crucial).
    3. Close the modal.
    4. Move the question to another collection
    5. Repeat step 2, now you'll see the source question on both the previous collection and the new collection.
    https://www.loom.com/share/3f0b588b4a634f17bf5dc2aa6e10ef6e
3. Clicking `x` button on the parameter source picker modal will show a weird error `API endpoint does not exist.`
    image
4. Not entirely sure if showing duplicate entries in the dropdown selection would confuse users or not. I know this could happen because we don't modify the source question. But as you could see, choosing one option also choosing other options which have the same value, so many FE could dedupe the option here to make the UX more streamline. What do you think?
    Screenshot 2022-12-23 at 11 16 22 AM

**Comment by ranquild** [comment]
@WiNloSt 
- 4 is by design atm
- 1 and 2 seems to bugs in the `DataPicker`. @kulyk could you take a look?
- I'll take a look at 3

**Comment by ranquild** [comment]
@WiNloSt 1-3 should be fixed atm

**Review by kulyk — APPROVED** [review]
LGTM, nice work

**Review by WiNloSt — COMMENTED** [review]
I didn't have a comprehensive review, but the code looks good to me.

**Pull request merged**